pyiceberg/table/__init__.py

```diff
@@ -1512,8 +1432,33 @@ class Move:
     other_field_id: Optional[int] = None
 
 
-class UpdateSchema:
-    _table: Optional[Table]
+U = TypeVar('U')
+
+
+class TableMetadataUpdate(ABC, Generic[U]):
+    _transaction: Transaction
+    _table_metadata: TableMetadata
+
+    def __init__(self, transaction: Transaction, table_metadata: TableMetadata) -> None:
+        self._transaction = transaction
+        self._table_metadata = table_metadata
```

Comment:
[Question] I am a little bit confused here. Why do we have both `transaction` and `table_metadata` instead of `transaction` only since table metadata is also stored in the transaction?

Reply:
That's a very good question, I don't think this is required, and might even be confusing 👍

Reply:
I think this was still in there from when you could do an update directly on a table itself. I've removed it and it is much cleaner now.